newrelic-agent/src/main/java/com/newrelic/agent/trace/RandomTransactionSampler.java

```diff
@@ -57,7 +58,7 @@ public List<TransactionTrace> harvest(String appName) {
         if (td == null) {
             return Collections.emptyList();
         }
-        if (td.getApplicationName() != appName) {
```

Comment:
Do a null safe comparison with `Objects.equals`

Reply:
There are some places where we do String comparison using `==` as a performance improvement.
In those cases, we make sure that we always use the same set of Strings so we know that if the Strings are going to be `equals`, then they are going to be the same reference.
Not a coding best practice, but it lessens the overhead of the agent.